x/archivedb/db.go

```diff
@@ -85,61 +89,35 @@ func (db *archiveDB) GetLastBlock(key []byte) ([]byte, uint64, error) {
 	return db.Get(key, db.currentHeight)
 }
 
+// Returns an object which implements the database.KeyValueReader trait for all
+// keys defined at the given height.
+func (db *archiveDB) GetHeightReader(height uint64) (dbHeightReader, error) {
+	if height > db.currentHeight || height == 0 {
```

Comment:
Do we need to error on `height == 0` ? I would think we can just return a reader that's expected to not contain any key value pairs?